src/PostgREST/Config.hs

```diff
@@ -240,6 +252,8 @@ readAppConfig cfgPath = do
         <*> (maybe [] (fmap encodeUtf8 . splitOnCommas) <$> optValue "raw-media-types")
         <*> pure Nothing
         <*> parseLogLevel "log-level"
+        <*> ((Just False ==) <$> optBool "tx-rollback-all")
+        <*> ((Just False ==) <$> optBool "tx-allow-override")
```

Comment:
Both of these default to False right?

I remember this part being confusing because of Hlint. Check this: https://github.com/ndmitchell/hlint/issues/970#issuecomment-643646635

Maybe newer versions don't show this hint.

Reply:
Uh, eh. I have no idea. This is not covered by the tests, I guess I'll need to add to the io-tests.

I just copied from a couple more lines above, I think and thought False must be False. But according to the link you're giving... I am completely lost now :D

Reply:
Actually when trying the `tx-rollback-all=true` while running the postgrest binary I don't get the `all ROLLBACK` behavior. When changing these to `(Just True ==)` I do get the behavior. Maybe this needs to be corrected?

Reply:
Hm. Ok. I had a brief look at the IO tests, but I don't think it's simple to add such a test there. What I would really like to have is a command line argument to call the postgrest binary with, which just exits and dumps the effectively loaded config. This could be useful for debugging and the io tests.

Reply:
Since this is **very** confusing, I added some **very** explicit comments on each of those boolean options to make that clear.

The IO test is something for later.

Reply:
Yes, **very** confusing. We should look at the latest hlint version and see if that still prevents us from using a `fromMaybe False`.

Reply:
That might be very useful! Related topic if we add an exit command line option (or env variable): It would also be useful to have an option for postgrest to exit once the dbstrucutre is loaded. That would allow us to benchmark the startup time (including in CI) as a basis for refactorings/improvements

Reply:
Those should definitely be command line options and not environment variables.

Something like `--dump-config` could work. Is there anything useful that we could output for the dbstructure? What if `--dump-structure` would output a json with all the query results?

Reply:
> Yes, **very** confusing. We should look at the latest hlint version and see if that still prevents us from using a `fromMaybe False`.

I just changed all of them to a `fromMaybe False` and hlint is not complaining. Maybe this is fixed already with what we have now. Let's see what CI says.

Reply:
Yes, that would be the relevant output. Will need to discuss in more detail at some other point I think, just wanted to flag that we might want to consider additional exit points if we add that command line option